test/e2e/allocator_test.go

```diff
@@ -80,7 +83,9 @@ func TestAllocator(t *testing.T) {
 		response, err = client.Post(requestURL, "application/json", bytes.NewBuffer(body))
 
 		if err != nil {
-			response.Body.Close() // nolint: errcheck
+			if response != nil {
+				response.Body.Close() // nolint: errcheck
```

Comment:
Ah! Small Go gotcha. response.Body.Close() should always get run if the response not nil. Usually `defer` is a nice way to do it.

Details, and examples: https://stackoverflow.com/questions/33238518/what-could-happen-if-i-dont-close-response-body-in-golang

Reply:
response.Body.Close() is happening after the poll. So this one was not necessary.

Reply:
Aha! Nice!